### PR TITLE
Feat administration

### PR DESCRIPTION
**Pull request by remcocgi**
connects to osgp/platform#189 connects to osgp/platform#190

### CODE REVIEW

**Comment by CLAAdmin** [comment]
[CLA assistant check](http://52.16.237.165:80/OSGP/Platform?pullRequest=252) <br/>You should sign our Contributor License Agreement in order to get your pull request merged.<br/>

File: osgp-adapter-domain-admin/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/target/
+/target/
+/target/
```
Comment (bvdzwet) [inline osgp-adapter-domain-admin/.gitignore]: This kind of changes should not be in a commit.

File: osgp-adapter-domain-core/.gitignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+/target/
+/target/
```
Comment (bvdzwet) [inline osgp-adapter-domain-core/.gitignore]: This kind of changes should not be in a commit.

File: osgp-adapter-domain-publiclighting/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+/target/
+/target/
```
Comment (bvdzwet) [inline osgp-adapter-domain-publiclighting/.gitignore]: This kind of changes should not be in a commit.

File: osgp-adapter-domain-smartmetering/.gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+/target/
+/target/
```
Comment (bvdzwet) [inline osgp-adapter-domain-smartmetering/.gitignore]: This kind of changes should not be in a commit.

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/services/ConfigurationService.java [hunk 5]
```diff
@@ -132,7 +134,33 @@ public void setAlarmNotifications(@Identification final String organisationIdent
                 deviceIdentification, alarmNotificationsDto), messageType);
     }
 
-    public void setActivityCalendar(@Identification final String organisationIdentification,
+    public void setAdministration(@Identification final String organisationIdentification,
```
Comment (bvdzwet) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/services/ConfigurationService.java]: If the device function is renamed (see comments in Shared about administration vs administrative state), the method should be named consistently. The parameter setAdministration should be focused on the contents of the message (the stuff actually being set) and would probably better be named administrativeState (nothing about getting/setting there).
The same sort of comments apply for the method getAdministration.

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/mapping/GetAdministrationConverter.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
+package com.alliander.osgp.adapter.domain.smartmetering.application.mapping;
+
+import ma.glasnost.orika.converter.BidirectionalConverter;
+import ma.glasnost.orika.metadata.Type;
+
+import com.alliander.osgp.dto.valueobjects.smartmetering.GetAdministration;
+
+public class GetAdministrationConverter extends
```
Comment (bvdzwet) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/mapping/GetAdministrationConverter.java]: The administration part of the name should probably be changed to administrative state.

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/mapping/SetAdministrationConverter.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
+package com.alliander.osgp.adapter.domain.smartmetering.application.mapping;
+
+import ma.glasnost.orika.converter.BidirectionalConverter;
+import ma.glasnost.orika.metadata.Type;
+
+import com.alliander.osgp.domain.core.valueobjects.smartmetering.SetAdministration;
+
+public class SetAdministrationConverter extends
```
Comment (bvdzwet) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/mapping/SetAdministrationConverter.java]: The administration part of the name should probably be changed to administrative state.

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/services/ConfigurationService.java [hunk 8]
```diff
@@ -132,7 +134,33 @@ public void setAlarmNotifications(@Identification final String organisationIdent
...
+            @Identification final String deviceIdentification, final String correlationUid,
+            final SetAdministration setAdministration, final String messageType) throws FunctionalException {
+
+        LOGGER.info("SetAdministration for organisationIdentification: {} for deviceIdentification: {}",
```
Comment (bvdzwet) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/services/ConfigurationService.java]: The administration part of the function name in the log statement should probably be changed to administrative state.

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/services/ConfigurationService.java [hunk 9]
```diff
@@ -132,7 +134,33 @@ public void setAlarmNotifications(@Identification final String organisationIdent
...
+
+        this.osgpCoreRequestMessageSender.send(new RequestMessage(correlationUid, organisationIdentification,
+                deviceIdentification, setAdministration), messageType);
+    }
+
+    public void getAdministration(@Identification final String organisationIdentification,
+            @Identification final String deviceIdentification, final String correlationUid,
+            final GetAdministration getAdministration, final String messageType) throws FunctionalException {
+
+        LOGGER.info("SetAdministration for organisationIdentification: {} for deviceIdentification: {}",
```
Comment (bvdzwet) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/services/ConfigurationService.java]: The administration part of the function name in the log statement should probably be changed to administrative state.

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/GetAdministrationResponseMessageProcessor.java [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+import org.springframework.stereotype.Component;
+
+import com.alliander.osgp.adapter.domain.smartmetering.application.services.ConfigurationService;
+import com.alliander.osgp.adapter.domain.smartmetering.infra.jms.core.OsgpCoreResponseMessageProcessor;
+import com.alliander.osgp.domain.core.valueobjects.DeviceFunction;
+import com.alliander.osgp.domain.core.valueobjects.smartmetering.GetAdministration;
+import com.alliander.osgp.shared.infra.jms.Constants;
+
+@Component("domainSmartMeteringGetAdministrationResponseMessageProcessor")
+public class GetAdministrationResponseMessageProcessor extends OsgpCoreResponseMessageProcessor {
```
Comment (bvdzwet) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/GetAdministrationResponseMessageProcessor.java]: The administration part of the name should probably be changed to administrative state.

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/GetAdministrationResponseMessageProcessor.java [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+    @Autowired
+    private ConfigurationService configurationService;
+
+    public GetAdministrationResponseMessageProcessor() {
+        super(DeviceFunction.GET_ADMINISTRATION);
+    }
+
+    @Override
+    public void processMessage(final ObjectMessage message) throws JMSException {
+        LOGGER.debug("Processing smart metering get administration response message");
```
Comment (bvdzwet) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/GetAdministrationResponseMessageProcessor.java]: The administration part of the function name in the log statement should probably be changed to administrative state.

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/mapping/GetAdministrationConverter.java [hunk 12]
```diff
@@ -0,0 +1,41 @@
+package com.alliander.osgp.adapter.domain.smartmetering.application.mapping;
```
Comment (bvdzwet) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/mapping/GetAdministrationConverter.java]: This file is missing a license header.

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/mapping/SetAdministrationConverter.java [hunk 13]
```diff
@@ -0,0 +1,43 @@
+package com.alliander.osgp.adapter.domain.smartmetering.application.mapping;
```
Comment (bvdzwet) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/application/mapping/SetAdministrationConverter.java]: This file is missing a license header.

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/GetAdministrationResponseMessageProcessor.java [hunk 14]
```diff
@@ -0,0 +1,69 @@
+package com.alliander.osgp.adapter.domain.smartmetering.infra.jms.core.messageprocessors;
```
Comment (bvdzwet) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/GetAdministrationResponseMessageProcessor.java]: This file is missing a license header.

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/SetAdministrationResponseMessageProcessor.java [hunk 15]
```diff
@@ -0,0 +1,68 @@
+package com.alliander.osgp.adapter.domain.smartmetering.infra.jms.core.messageprocessors;
```
Comment (bvdzwet) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/SetAdministrationResponseMessageProcessor.java]: This file is missing a license header.

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/SetAdministrationResponseMessageProcessor.java [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+import org.springframework.stereotype.Component;
+
+import com.alliander.osgp.adapter.domain.smartmetering.application.services.ConfigurationService;
+import com.alliander.osgp.adapter.domain.smartmetering.infra.jms.core.OsgpCoreResponseMessageProcessor;
+import com.alliander.osgp.domain.core.valueobjects.DeviceFunction;
+import com.alliander.osgp.domain.core.valueobjects.smartmetering.SetAdministration;
+import com.alliander.osgp.shared.infra.jms.Constants;
+
+@Component("domainSmartMeteringSetAdministrationResponseMessageProcessor")
+public class SetAdministrationResponseMessageProcessor extends OsgpCoreResponseMessageProcessor {
```
Comment (bvdzwet) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/SetAdministrationResponseMessageProcessor.java]: The administration part of the name should probably be changed to administrative state.

File: osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/SetAdministrationResponseMessageProcessor.java [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+    @Autowired
+    private ConfigurationService configurationService;
+
+    public SetAdministrationResponseMessageProcessor() {
+        super(DeviceFunction.SET_ADMINISTRATION);
+    }
+
+    @Override
+    public void processMessage(final ObjectMessage message) throws JMSException {
+        LOGGER.debug("Processing smart metering set administration response message");
```
Comment (bvdzwet) [inline osgp-adapter-domain-smartmetering/src/main/java/com/alliander/osgp/adapter/domain/smartmetering/infra/jms/core/messageprocessors/SetAdministrationResponseMessageProcessor.java]: The administration part of the function name in the log statement should probably be changed to administrative state.

File: osgp-adapter-domain-tariffswitching/.gitignore [hunk 18]
```diff
@@ -0,0 +1 @@
+/target/
```
Comment (bvdzwet) [inline osgp-adapter-domain-tariffswitching/.gitignore]: This kind of changes should not be in a commit.

File: osgp-adapter-ws-admin/.gitignore [hunk 19]
```diff
@@ -0,0 +1,2 @@
+/target/
+/target/
```
Comment (bvdzwet) [inline osgp-adapter-ws-admin/.gitignore]: This kind of changes should not be in a commit.

File: osgp-adapter-ws-core/.gitignore [hunk 20]
```diff
@@ -0,0 +1,2 @@
+/target/
+/target/
```
Comment (bvdzwet) [inline osgp-adapter-ws-core/.gitignore]: This kind of changes should not be in a commit.

File: osgp-adapter-ws-publiclighting/.gitignore [hunk 21]
```diff
@@ -0,0 +1 @@
+/target/
```
Comment (bvdzwet) [inline osgp-adapter-ws-publiclighting/.gitignore]: This kind of changes should not be in a commit.

File: osgp-adapter-ws-shared-db/.gitignore [hunk 22]
```diff
@@ -0,0 +1,3 @@
+/target/
+/target/
+/target/
```
Comment (bvdzwet) [inline osgp-adapter-ws-shared-db/.gitignore]: This kind of changes should not be in a commit.

File: osgp-adapter-ws-shared/.gitignore [hunk 23]
```diff
@@ -0,0 +1,3 @@
+/target/
+/target/
+/target/
```
Comment (bvdzwet) [inline osgp-adapter-ws-shared/.gitignore]: This kind of changes should not be in a commit.

File: osgp-adapter-ws-smartmetering/.gitignore [hunk 24]
```diff
@@ -0,0 +1 @@
+/target/
```
Comment (bvdzwet) [inline osgp-adapter-ws-smartmetering/.gitignore]: This kind of changes should not be in a commit.

File: osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/mapping/GetAdministrationConverter.java [hunk 25]
```diff
@@ -0,0 +1,42 @@
+package com.alliander.osgp.adapter.ws.smartmetering.application.mapping;
```
Comment (bvdzwet) [inline osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/mapping/GetAdministrationConverter.java]: This file is missing a license header.

File: osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/mapping/GetAdministrationConverter.java [hunk 26]
```diff
@@ -0,0 +1,42 @@
...
+
+import ma.glasnost.orika.converter.BidirectionalConverter;
+import ma.glasnost.orika.metadata.Type;
+
+import com.alliander.osgp.domain.core.valueobjects.smartmetering.GetAdministration;
+
+public class GetAdministrationConverter
```
Comment (bvdzwet) [inline osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/mapping/GetAdministrationConverter.java]: The administration part of the name should probably be changed to administrative state.

File: osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/mapping/SetAdministrationConverter.java [hunk 27]
```diff
@@ -0,0 +1,42 @@
+package com.alliander.osgp.adapter.ws.smartmetering.application.mapping;
+
+import ma.glasnost.orika.converter.BidirectionalConverter;
+import ma.glasnost.orika.metadata.Type;
+
+import com.alliander.osgp.adapter.ws.schema.smartmetering.configuration.SetAdministrationRequest;
+import com.alliander.osgp.domain.core.valueobjects.smartmetering.SetAdministration;
+
+public class SetAdministrationConverter extends BidirectionalConverter<SetAdministration, SetAdministrationRequest> {
```
Comment (bvdzwet) [inline osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/mapping/SetAdministrationConverter.java]: The administration part of the name should probably be changed to administrative state.

File: osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/mapping/SetAdministrationConverter.java [hunk 28]
```diff
@@ -0,0 +1,42 @@
+package com.alliander.osgp.adapter.ws.smartmetering.application.mapping;
```
Comment (bvdzwet) [inline osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/application/mapping/SetAdministrationConverter.java]: This file is missing a license header.

File: osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/endpoints/SmartMeteringConfigurationEndpoint.java [hunk 29]
```diff
@@ -61,6 +65,49 @@
     public SmartMeteringConfigurationEndpoint() {
     }
 
+    @PayloadRoot(localPart = "SetAdministrationRequest", namespace = SMARTMETER_CONFIGURATION_NAMESPACE)
+    @ResponsePayload
+    public SetAdministrationResponse setAdministration(
```
Comment (bvdzwet) [inline osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/endpoints/SmartMeteringConfigurationEndpoint.java]: Administration should be called something like AdministrativeState.

File: osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/infra/jms/SmartMeteringRequestMessageType.java [hunk 30]
```diff
@@ -21,6 +21,8 @@
     REQUEST_SPECIAL_DAYS,
     SET_ALARM_NOTIFICATIONS,
     SET_CONFIGURATION_OBJECT,
+    SET_ADMINISTRATION,
```
Comment (bvdzwet) [inline osgp-adapter-ws-smartmetering/src/main/java/com/alliander/osgp/adapter/ws/smartmetering/infra/jms/SmartMeteringRequestMessageType.java]: Administration should probably be named something like AdministrativeState.

File: osgp-adapter-ws-smartmetering/src/main/webapp/WEB-INF/wsdl/smartmetering/SmartMeteringConfiguration.wsdl [hunk 31]
```diff
@@ -74,6 +74,21 @@
     	<wsdl:part element="smconf:RetrieveSetActivityCalendarResultResponse" name="RetrieveSetActivityCalendarResultResponse" />
     </wsdl:message>
 
+
```
Comment (bvdzwet) [inline osgp-adapter-ws-smartmetering/src/main/webapp/WEB-INF/wsdl/smartmetering/SmartMeteringConfiguration.wsdl]: Administration should probably be named something like AdministrativeState.

File: osgp-adapter-ws-smartmetering/src/main/webapp/WEB-INF/wsdl/smartmetering/schemas/sm-configuration.xsd [hunk 32]
```diff
@@ -10,6 +10,40 @@
 	<xsd:import namespace="http://www.alliander.com/schemas/osgp/common/2014/10"
 		schemaLocation="common.xsd" />
 
+	<xsd:element name="GetAdministrationRequest">
```
Comment (bvdzwet) [inline osgp-adapter-ws-smartmetering/src/main/webapp/WEB-INF/wsdl/smartmetering/schemas/sm-configuration.xsd]: Administration should probably be named something like AdministrativeState.

File: osgp-adapter-ws-smartmetering/src/main/webapp/WEB-INF/wsdl/smartmetering/schemas/sm-configuration.xsd [hunk 33]
```diff
@@ -10,6 +10,40 @@
...
+				<xsd:element name="Enabled" type="xsd:boolean" />
+			</xsd:sequence>
+		</xsd:complexType>
+	</xsd:element>
+
+	<xsd:element name="SetAdministrationRequest">
+		<xsd:complexType>
+			<xsd:sequence>
+				<xsd:element name="DeviceIdentification" type="common:DeviceIdentification" />
+				<xsd:element name="Enabled" type="xsd:boolean" />
```
Comment (bvdzwet) [inline osgp-adapter-ws-smartmetering/src/main/webapp/WEB-INF/wsdl/smartmetering/schemas/sm-configuration.xsd]: I suspect this should reflect the triple state from the DSMR specs for Administrative In/Out.

File: osgp-adapter-ws-smartmetering/src/main/webapp/WEB-INF/wsdl/smartmetering/schemas/sm-configuration.xsd [hunk 34]
```diff
@@ -10,6 +10,40 @@
...
+				<xsd:element name="DeviceIdentification" type="common:DeviceIdentification" />
+			</xsd:sequence>
+		</xsd:complexType>
+	</xsd:element>
+	
+	<xsd:element name="GetAdministrationResponse">
+		<xsd:complexType>
+			<xsd:sequence>
+				<xsd:element name="AsyncResponse" type="common:AsyncResponse" />
+				<xsd:element name="Enabled" type="xsd:boolean" />
```
Comment (bvdzwet) [inline osgp-adapter-ws-smartmetering/src/main/webapp/WEB-INF/wsdl/smartmetering/schemas/sm-configuration.xsd]: A response is either an AsyncResponse, or a response containing the actual data requested (and not a mix of both).
For requests that communicate with a smart meter an async response is returned, and another request should be added to retrieve the actual result.

File: osgp-adapter-ws-tariffswitching/.gitignore [hunk 35]
```diff
@@ -0,0 +1 @@
+/target/
```
Comment (bvdzwet) [inline osgp-adapter-ws-tariffswitching/.gitignore]: This kind of changes should not be in a commit.

File: osgp-core/.gitignore [hunk 36]
```diff
@@ -0,0 +1,2 @@
+/target/
+/target/
```
Comment (bvdzwet) [inline osgp-core/.gitignore]: This kind of changes should not be in a commit.

File: osgp-domain-core/.gitignore [hunk 37]
```diff
@@ -0,0 +1,2 @@
+/target/
+/target/
```
Comment (bvdzwet) [inline osgp-domain-core/.gitignore]: This kind of changes should not be in a commit.

File: osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/DeviceFunction.java [hunk 38]
```diff
@@ -42,6 +42,8 @@
     REQUEST_SPECIAL_DAYS,
     SET_ALARM_NOTIFICATIONS,
     SET_CONFIGURATION_OBJECT,
+    SET_ADMINISTRATION,
+    GET_ADMINISTRATION,
```
Comment (bvdzwet) [inline osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/DeviceFunction.java]: Administration should probably be named something like AdministrativeState.

File: osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/smartmetering/SetAdministration.java [hunk 39]
```diff
@@ -0,0 +1,29 @@
...
+
+    public boolean isEnabled() {
+        return this.enabled;
+    }
+
+    public void setEnabled(final boolean enabled) {
+        this.enabled = enabled;
+    }
+
+    public String getDeviceIdentification() {
```
Comment (bvdzwet) [inline osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/smartmetering/SetAdministration.java]: DeviceIdentification could be removed I think. For instance with setting of alarm notifications it is not included, and that appears to work ok.

File: osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/smartmetering/SetAdministration.java [hunk 40]
```diff
@@ -0,0 +1,29 @@
...
+
+    private boolean enabled;
+
+    private String deviceIdentification;
+
+    public boolean isEnabled() {
+        return this.enabled;
+    }
+
+    public void setEnabled(final boolean enabled) {
```
Comment (bvdzwet) [inline osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/valueobjects/smartmetering/SetAdministration.java]: Value objects should be immutable.

File: osgp-logging/.gitignore [hunk 41]
```diff
@@ -0,0 +1 @@
+/target/
```
Comment (bvdzwet) [inline osgp-logging/.gitignore]: This kind of changes should not be in a commit.

**Pull request merged**